### PR TITLE
Use logging not log

### PR DESCRIPTION
**Pull request by johanlundberg**
### All Submissions:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Have you added an explanation of what problem you are trying to solve with this PR?
* [ ] Have you added information on what your changes do and why you chose this as your solution?
* [ ] Have you written new tests for your changes?
* [x] Does your submission pass tests?
* [x] This project follows PEP8 style guide. Have you run your code against the 'flake8' linter?

### CODE REVIEW

**Comment by leifj** [comment]
I don't think this is quite right. I think the intention is to grab a log instance like in __init__.py and use that. This enables filtering of loggs (esp debugging) on a per-file/module basis.

**Comment by johanlundberg** [comment]
Ok, right now the program crashes is there is an exception due to there is nothing named log. I will have another look at it.

**Comment by c00kiemon5ter** [comment]
All of what you say is correct. There are multiple things here
- There is nothing named `log`, so the exception handler will crash (the problem)
- The rest of the module used `logging` directly, so we replace `log` with `logging` to fix the problem (the solution)
- The proper way to use a logger, would be to _get_ it, set it up and expose that as configuration (what Leif suggested)

I think we can keep those separate. Fix the problem with this PR and address logging in the whole module in another PR (or at least another commit).

File: src/xmlsec/__init__.py [hunk 1]
```diff
@@ -6,9 +6,12 @@
 __author__ = 'leifj'
 
 import six
+import logging
+# Instantiate log here to avoid circular dependency
+log = logging.getLogger('xmlsec')
+
```
Comment (c00kiemon5ter) [inline src/xmlsec/__init__.py]: This whole file does not need to change. If we are to change something it should be the logger name.

```python
log = logging.getLogger(__name__)
```

**Comment by johanlundberg** [comment]
I just pushed a commit that import log from __init__.py but I am not very pleased with it as I had to move the instantiation of log to the top of the import as crypto is being imported in __init__.py.

File: src/xmlsec/crypto.py [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import io
 import os
 import base64
-import logging
 import threading
 import six
 from six.moves import xrange
-from xmlsec import constants
+from xmlsec import constants, log
```
Comment (c00kiemon5ter) [inline src/xmlsec/crypto.py]: Do not do this. Instead grab a new logger instance from `logging`

```python
...
import logging


log = logging.getLogger(__name__)
```

This means that we can configure `xmlsec` differently from `xmlsec.crypto` (these are the loggers' names).

Comment (johanlundberg) [reply src/xmlsec/crypto.py]: Ok, I might have misunderstood Leif, I agree that this is better.

**Pull request merged**